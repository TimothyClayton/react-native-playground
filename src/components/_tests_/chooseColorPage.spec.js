import React from 'react'
import { Button } from 'react-native'
import ChooseColorPage from '../ChooseColorPage'
import { shallow, simulate } from 'enzyme'

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Colors from '../../state/Colors';

Enzyme.configure({ adapter: new Adapter() });

it('Renders correctly', () => {
  const wrapper = shallow(
    <Button key={1} title={'BLUE'} color={'str'} onPress={ jest.fn() } />
    // <ChooseColorPage colorChanged={ jest.fn() } navigation={{ goBack: jest.fn() }} />
  )
  // wrapper.find(Button).simulate('click')
  expect(wrapper).toMatchSnapshot();
})

it('shows all the colors', () => {
  const wrapper = shallow(<ChooseColorPage />)
  console.log(wrapper)
  // expect()
})
