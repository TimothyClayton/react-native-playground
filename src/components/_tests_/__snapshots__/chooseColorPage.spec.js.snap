// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    color="str"
    onPress={[MockFunction]}
    title="BLUE"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "accessibilityLabel": undefined,
      "accessibilityRole": "button",
      "accessibilityStates": Array [],
      "activeOpacity": 0.2,
      "children": <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "str",
              },
            ]
          }
        >
          BLUE
        </Text>
      </View>,
      "disabled": undefined,
      "hasTVPreferredFocus": undefined,
      "onPress": [MockFunction],
      "testID": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "str",
              },
            ]
          }
        >
          BLUE
        </Text>,
        "style": Array [
          Object {},
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "BLUE",
          "disabled": undefined,
          "style": Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "str",
            },
          ],
        },
        "ref": null,
        "rendered": "BLUE",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessibilityLabel": undefined,
        "accessibilityRole": "button",
        "accessibilityStates": Array [],
        "activeOpacity": 0.2,
        "children": <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "str",
                },
              ]
            }
          >
            BLUE
          </Text>
        </View>,
        "disabled": undefined,
        "hasTVPreferredFocus": undefined,
        "onPress": [MockFunction],
        "testID": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "str",
                },
              ]
            }
          >
            BLUE
          </Text>,
          "style": Array [
            Object {},
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "BLUE",
            "disabled": undefined,
            "style": Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "str",
              },
            ],
          },
          "ref": null,
          "rendered": "BLUE",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
